build_tools/docker/manage_images.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Manages IREE Docker image definitions.
+
+Includes information on their dependency graph and GCR URL.
+
+Example usage:
+
+Rebuild the cmake image and all images that transitiviely on depend on it,
+tagging them with `latest`:
+  python3 build_tools/docker/manage_images.py --build --image cmake
+
+Print out output for rebuilding the cmake image and all images that
+transitiviely on depend on it, but don't take side-effecting actions:
+  python3 build_tools/docker/manage_images.py --build --image cmake --dry-run
+
+Push all `prod` images to GCR:
+  python3 build_tools/docker/manage_images.py --push --tag prod --images all
+
+Rebuild and push all images and update references to them in the repository:
```

Comment:
Should we include the info about when and why you'd want to do this in the docker README?

Reply:
I have some README updates in my other PR, but yes we should. I'm going to defer that for later to minimize the conflicts here.